haystack_experimental/components/generators/chat/openai.py

```diff
@@ -0,0 +1,20 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import haystack.components.generators.chat.openai
+from haystack_experimental.dataclasses.chat_message import ChatMessage
+from haystack import component
+
+# Monkey patch the Haystack ChatMessage class with the experimental one. By doing so, we can use the new 
+# `to_openai_dict_format` method, allowing multimodal chat messages.
+haystack.components.generators.chat.openai.ChatMessage = ChatMessage
+
+@component
+class OpenAIChatGenerator(haystack.components.generators.chat.openai.OpenAIChatGenerator):
```

Comment:
Just curious to understand: this is here because there's currently no OpenAIChatGenerator on experimental? do we really need it, couldn't we import it from haystack core?

Reply:
It's a bit tricky, but doing `from haystack_experimental.components.generators.chat import OpenAIChatGenerator` ensures that the line before (where we monkey-patch `ChatMessage` with its experimental counterpart) is always executed.

Reply:
ok, got it - I see the point now 👍🏽